source/common/grpc/async_client_impl.cc

```diff
@@ -63,7 +63,11 @@ AsyncStreamImpl::AsyncStreamImpl(AsyncClientImpl& parent, absl::string_view serv
                                  absl::string_view method_name, RawAsyncStreamCallbacks& callbacks,
                                  const Http::AsyncClient::StreamOptions& options)
     : parent_(parent), service_full_name_(service_full_name), method_name_(method_name),
-      callbacks_(callbacks), options_(options) {}
+      callbacks_(callbacks), options_(options) {
+  if (options_.parent_context.stream_info != nullptr) {
+    metadata_parser_ = Router::HeaderParser::configure(parent_.initial_metadata_, false);
```

Comment:
Can this be built at client-level so we don't need to do it at every stream creation?